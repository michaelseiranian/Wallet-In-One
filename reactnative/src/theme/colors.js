// Colours used across different components of the app when 'light' theme is enabled in settings.
export const lightColors = {
    background: 'white',
    text: 'black',
    primary: 'purple',  // Main theme colour of the app
    secondary: 'blue',
    stock_account: 'white', // Used for the background of the main stocks screen
};

// Colours used across different components of the app when 'dark' theme is enabled in settings.
export const darkColors = {
    background: '#242526',
    text: 'white',
    primary: 'purple',  // Main theme colour of the app
    secondary: 'blue',
    stock_account: '#181818', // Used for the background of the main stocks screen
};