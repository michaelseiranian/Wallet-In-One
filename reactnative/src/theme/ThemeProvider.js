import React, {createContext, useEffect, useContext, useState, AsyncStorage} from 'react';
import {lightColors, darkColors} from './colors';
import {useColorScheme} from 'react-native';
export const ThemeContext = createContext({
    dark: false,
    colors: lightColors,
    setScheme: () => {},
    update: () => {}
});

import * as SecureStore from 'expo-secure-store';

export const ThemeProvider = props => {
    const defaultColorScheme = useColorScheme(); // colorScheme is either 'dark', 'light' or 'null'
    const [isDark, setIsDark] = useState(false);

    async function loadDarkModeSettings(value = 'default') {
        // colorScheme is either 'dark', 'light' or 'null'
        var darkSetting;
        if (value === "default"){
            darkSetting = await SecureStore.getItemAsync('darkModeSettings');
        }
        else{
            darkSetting = value;
        }

        console.log('settings, default: ', darkSetting, defaultColorScheme,value)

        if (darkSetting === 'true') {
            setIsDark(true)
        }
        else if (darkSetting === 'false') {
            setIsDark(false)
        }
        else if (defaultColorScheme == 'dark') {
            setIsDark(true)
        }
        else if (defaultColorScheme == 'light') {
            setIsDark(false)
        }
        else {
            setIsDark(false)
        }
    }

    // Load Dark Mode Setting
    useEffect(() => {    
        loadDarkModeSettings();
    }, [defaultColorScheme]);

    const defaultTheme = {
        dark: isDark,
        colors: isDark ? darkColors : lightColors,
        setScheme: scheme => setIsDark(scheme === 'dark'),
        update: loadDarkModeSettings,
    };
    return (
        <ThemeContext.Provider value={defaultTheme}>
            {props.children}
        </ThemeContext.Provider>
    );
};

// creating a custom hook for accessing all the values

export const useTheme = () => useContext(ThemeContext);