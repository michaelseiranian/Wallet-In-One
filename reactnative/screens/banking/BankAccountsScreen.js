import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View, ScrollView, Button, TextInput, Alert, FlatList , TouchableOpacity, Image} from 'react-native';
import { useState, useEffect } from 'react';
import { auth_get} from '../../authentication'
import { useIsFocused } from '@react-navigation/native';
import Loading from './Loading'
import { useTheme } from 'reactnative/src/theme/ThemeProvider'
import {styles} from 'reactnative/screens/All_Styles.style.js'
import AntDesign from 'react-native-vector-icons/AntDesign';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import ConditionalModal from '../Modal';

export default function BankAccountsScreen({ navigation }) {
  const [ isLoading, setIsLoading ] = useState(true)
  const [ bankData, setBankData ] = useState([])
  const isFocused = useIsFocused();
  const {dark, colors, setScheme} = useTheme();
  const [modalVisible, setModalVisible] = useState(false);

  const deleteAccount = async (id) => {
    console.log(`delete_account /banking/delete_account/${id}/`)
    setIsLoading(true)
    await auth_get(`/banking/delete_account/${id}/`)
    fetchData();
  }

  const fetchData = async () => {
    console.log('fetch bank accounts data')
    setIsLoading(true)
    const response = await auth_get('/banking/user_accounts/')
    console.log('status', response.status)
    if (response.status == 200){
        setIsLoading(false)
        setBankData(response.body)
    }
  }

  useEffect(() =>{
    if(isFocused != false){fetchData()}
  }, [isFocused])

  const stylesInternal = StyleSheet.create({
    container: {
      width: '100%',
      paddingLeft: 20,
      paddingRight: 20,
      paddingBottom: 0,
      paddingTop: 10,
      borderRadius: 5,
      borderColor: dark ? colors.background : '#ddd',
      overflow: 'hidden',
    },
    row:{
      flexDirection: 'row',
      alignItems: 'flex-start',
    },
    item:{
      padding: 20,
      borderRadius: 10,
      marginBottom: 10,
    },
    item_error:{
      borderColor: 'red',
      borderWidth: 4,
    },
    name:{
      color: "white",
      fontWeight: 'bold',
      fontSize: 21,
    },
    amount:{
      color: "white",
      fontSize: 18,
    },
    iban:{
      color: 'rgba(255,255,255,0.5)',
      fontSize: 10,
      marginBottom: 3,
    },
    closeButton: {
      position: 'absolute',
      top: 10,
      right: 10,
      backgroundColor: 'red',
      width: 27,
      height: 27,
      borderRadius: 15,
      alignItems: 'center',
      justifyContent: 'center',
    },
    closeButton2: {
      position: 'absolute',
      bottom: 20,
      right: 10,
      backgroundColor: '#ecba1d',
      borderRadius: 15,
      width: 27,
      height: 27,
      alignItems: 'center',
      justifyContent: 'center',
    },
    closeButtonText: {
      color: 'white',
      fontSize: 20,
      fontWeight: 'bold',
    },
    button: {
      padding: 10,
      borderRadius: 10,
      marginVertical: 4,
      marginHorizontal: 8,
      backgroundColor: colors.primary,
      alignSelf: 'center',
    },
    buttonText: {
      textAlign: 'center',
      fontWeight: 'bold',
      color: colors.text,
    },
  });
  
  

  if (isLoading){
    return <Loading/>
}

  return (
    <View style={{flex:1, backgroundColor: colors.background}} >

        <View style={{flexDirection:'row', justifyContent: 'center'}}>
          <TouchableOpacity
            style={stylesInternal.button}
            onPress={() => navigation.navigate('Bank Insights')}
          >
            <Text style={stylesInternal.buttonText}>Bank Insights</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={stylesInternal.button}
            onPress={() => navigation.navigate("All Bank Transactions")}
          >
            <Text style={stylesInternal.buttonText}>All Transactions</Text>
          </TouchableOpacity>
        </View>

              <View style={[stylesInternal.container]}>
                  <FlatList data={bankData} renderItem={({item, index}) =>{
                      return ( 
                        BankAccount(item, index)) 
                      }}
                      ListEmptyComponent={<Text style={styles(dark, colors).text}>{'\nYou have no bank accounts\n'}</Text>}
                  />
              </View>
      </View>
  );

  function BankAccount(item, index) {
    var mainStyle = [stylesInternal.item, { backgroundColor: item.color }]
    if (item.disabled){
      mainStyle.push(stylesInternal.item_error)
    }

    return (
      <>
      <TouchableOpacity testID="account" style={mainStyle} key={index} onPress={ () => {  navigation.navigate('Bank Transactions', { accountID: item.id })}}>
        <View style={stylesInternal.row}>
          <Image
            source={{ uri: item.institution_logo }}
            style={{ width: 70, height: 70, marginRight: 10, resizeMode: 'contain', borderRadius: 10, backgroundColor: 'white' }} />
          <View style={{ borderRadius: 10 }}>
            <Text style={stylesInternal.name}>{item.institution_name}</Text>
            <Text style={stylesInternal.iban}>{item.iban}</Text>
            <Text style={stylesInternal.amount}>{item.balance.string}</Text>
          </View>
        </View>
      </TouchableOpacity>
      <TouchableOpacity testID="close1" style={stylesInternal.closeButton} onPress={() => setModalVisible(true)}>
        <FontAwesome style={stylesInternal.closeButtonText} name="close" size= {20}/>
      </TouchableOpacity>

      <ConditionalModal
        headerText={"Remove Your Bank Account"}
        bodyText={"Are you sure you want to remove your bank account?"}
        visible={modalVisible}
        onEvent={() => deleteAccount(item.id)}
        onClose={() => setModalVisible(false)}
      />

      {item.disabled?(
      <TouchableOpacity testID="close2" style={stylesInternal.closeButton2} onPress={() => Alert.alert('Warning','This account is not connected anymore. Please delete and readd this account.') }>
        <FontAwesome style={stylesInternal.closeButtonText} name="exclamation" size= {20}/>
      </TouchableOpacity>):(<></>)}
      
    </>

)
  }
}