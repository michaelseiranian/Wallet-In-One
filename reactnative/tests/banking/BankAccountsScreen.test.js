import React from 'react';
import { renderHook, render, screen, fireEvent, act, within, waitFor} from '@testing-library/react-native';
import BankAccountsScreen from '../../screens/banking/BankAccountsScreen'
import { useTheme } from 'reactnative/src/theme/ThemeProvider';
import * as SecureStore from "expo-secure-store";

jest.mock("expo-secure-store");
jest.mock('reactnative/src/theme/ThemeProvider');

SecureStore.getItemAsync.mockReset();

global.fetch =  jest.fn( async (api, data ) => {
    var response = [
        {
            "id": "1",
            "requisition_id": "None",
            "last_update": "2023-03-17T13:32:01.539829Z",
            "iban": "1234",
            "institution_id": "REVOLUT_REVOGB21",
            "institution_name": "Revolut",
            "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
            "color": "#18A7F7",
            "disabled": false,
            "user": 1,
            "balance": {
                "string": "£100",
                "currency": "GBP",
                "amount": "100"
            }
        },
        {
            "id": "2",
            "requisition_id": "None",
            "last_update": "2023-03-17T13:32:01.539829Z",
            "iban": "1234",
            "institution_id": "REVOLUT_REVOGB21",
            "institution_name": "Revolut",
            "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
            "color": "#18A7F7",
            "disabled": true,
            "user": 1,
            "balance": {
                "string": "£200",
                "currency": "GBP",
                "amount": "200"
            }
        },
    ]

    return Promise.resolve({
        status: 200, json: () => response
    })

})

describe('<BankAccountsScreen />', () => {
    it('snapshot and button test', async () => {
        useTheme.mockReturnValue({
            dark: true,
            colors: {
              text: '#000',
              background: '#fff',
              primary: '#0af',
            },
            setScheme: jest.fn(),
            update: jest.fn(),
          });

        const navigate = jest.fn();
        snapshot = render(<BankAccountsScreen navigation={{ navigate }} />);

        await waitFor( () => {
            const activityIndicator = screen.UNSAFE_queryByType('ActivityIndicator');
            expect(activityIndicator).toBeNull();
        })

        // Test button clicks
        await fireEvent.press(await screen.getByText('Bank Insights'));
        expect(navigate).toHaveBeenCalledWith('Bank Insights');

        await fireEvent.press(await screen.getByText('All Transactions'));
        await fireEvent.press(await screen.getAllByTestId('account')[0]);
        expect(navigate).toHaveBeenCalledWith('All Bank Transactions');
        
        await fireEvent.press(await screen.getAllByTestId('close1')[0]);
        await fireEvent.press(await screen.getByTestId('close2'));
        
        //const model = screen.UNSAFE_queryByType('Model');   
        expect(snapshot).toMatchSnapshot();
    })

    it('test pressing button 1 on model', async () => {
        useTheme.mockReturnValue({
            dark: false,
            colors: {
              text: '#000',
              background: '#fff',
              primary: '#0af',
            },
            setScheme: jest.fn(),
            update: jest.fn(),
          });

        const navigate = jest.fn();
        snapshot = render(<BankAccountsScreen navigation={{ navigate }} />);

        await waitFor( () => {
            const activityIndicator = screen.UNSAFE_queryByType('ActivityIndicator');
            expect(activityIndicator).toBeNull();
        })
        await fireEvent.press(await screen.getAllByTestId('close1')[0]);
        await fireEvent.press(await screen.getAllByTestId('pressable1')[0]);
    })

    it('test pressing button 2 on model', async () => {

        useTheme.mockReturnValue({
            dark: false,
            colors: {
              text: '#000',
              background: '#fff',
              primary: '#0af',
            },
            setScheme: jest.fn(),
            update: jest.fn(),
          });

        const navigate = jest.fn();
        snapshot = render(<BankAccountsScreen navigation={{ navigate }} />);

        await waitFor( () => {
            const activityIndicator = screen.UNSAFE_queryByType('ActivityIndicator');
            expect(activityIndicator).toBeNull();
        })
        await fireEvent.press(await screen.getAllByTestId('close1')[0]);
        await fireEvent.press(await screen.getAllByTestId('pressable2')[0]);
    })
});