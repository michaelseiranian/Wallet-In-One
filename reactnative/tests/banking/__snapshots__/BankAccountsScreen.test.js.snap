// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankAccountsScreen /> snapshot and button test 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#0af",
          "borderRadius": 10,
          "marginHorizontal": 8,
          "marginVertical": 4,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Bank Insights
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#0af",
          "borderRadius": 10,
          "marginHorizontal": 8,
          "marginVertical": 4,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        All Transactions
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "borderColor": "#fff",
          "borderRadius": 5,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <Text
          style={
            {
              "color": "#000",
            }
          }
        >
          
You have no bank accounts

        </Text>
      }
      data={
        [
          {
            "balance": {
              "amount": "100",
              "currency": "GBP",
              "string": "£100",
            },
            "color": "#18A7F7",
            "disabled": false,
            "iban": "1234",
            "id": "1",
            "institution_id": "REVOLUT_REVOGB21",
            "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
            "institution_name": "Revolut",
            "last_update": "2023-03-17T13:32:01.539829Z",
            "requisition_id": "None",
            "user": 1,
          },
          {
            "balance": {
              "amount": "200",
              "currency": "GBP",
              "string": "£200",
            },
            "color": "#18A7F7",
            "disabled": true,
            "iban": "1234",
            "id": "2",
            "institution_id": "REVOLUT_REVOGB21",
            "institution_logo": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
            "institution_name": "Revolut",
            "last_update": "2023-03-17T13:32:01.539829Z",
            "requisition_id": "None",
            "user": 1,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#18A7F7",
                "borderRadius": 10,
                "marginBottom": 10,
                "opacity": 1,
                "padding": 20,
              }
            }
            testID="account"
          >
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
                  }
                }
                style={
                  {
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "height": 70,
                    "marginRight": 10,
                    "resizeMode": "contain",
                    "width": 70,
                  }
                }
              />
              <View
                style={
                  {
                    "borderRadius": 10,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 21,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Revolut
                </Text>
                <Text
                  style={
                    {
                      "color": "rgba(255,255,255,0.5)",
                      "fontSize": 10,
                      "marginBottom": 3,
                    }
                  }
                >
                  1234
                </Text>
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                >
                  £100
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 15,
                "height": 27,
                "justifyContent": "center",
                "opacity": 1,
                "position": "absolute",
                "right": 10,
                "top": 10,
                "width": 27,
              }
            }
            testID="close1"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={true}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderBottomColor": "gray",
                    "borderBottomWidth": 1,
                    "borderRadius": 20,
                    "elevation": 5,
                    "marginHorizontal": 20,
                    "marginVertical": "20%",
                    "padding": 40,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 4,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#000",
                      "fontSize": 22,
                      "fontWeight": "bold",
                      "marginBottom": 10,
                    }
                  }
                >
                  Remove Your Bank Account
                </Text>
                <View
                  style={
                    {
                      "backgroundColor": "lightgray",
                      "height": 1,
                      "marginVertical": 10,
                      "width": "100%",
                    }
                  }
                />
                <Text
                  style={
                    {
                      "color": "#000",
                      "fontSize": 18,
                      "marginVertical": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  Are you sure you want to remove your bank account?
                </Text>
                <View
                  style={
                    {
                      "backgroundColor": "lightgray",
                      "height": 1,
                      "marginVertical": 10,
                      "width": "100%",
                    }
                  }
                />
                <View
                  style={
                    {
                      "alignItems": "flex-end",
                      "flexDirection": "row",
                      "justifyContent": "space-around",
                      "marginTop": 20,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "borderRadius": 10,
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        },
                        {
                          "backgroundColor": "#f44336",
                        },
                      ]
                    }
                    testID="pressable1"
                  >
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Yes
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "borderRadius": 10,
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        },
                        {
                          "backgroundColor": "#1E90FF",
                        },
                      ]
                    }
                    testID="pressable2"
                  >
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      No
                    </Text>
                  </View>
                </View>
              </View>
            </Modal>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#18A7F7",
                "borderColor": "red",
                "borderRadius": 10,
                "borderWidth": 4,
                "marginBottom": 10,
                "opacity": 1,
                "padding": 20,
              }
            }
            testID="account"
          >
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://cdn.nordigen.com/ais/REVOLUT_REVOGB21.png",
                  }
                }
                style={
                  {
                    "backgroundColor": "white",
                    "borderRadius": 10,
                    "height": 70,
                    "marginRight": 10,
                    "resizeMode": "contain",
                    "width": 70,
                  }
                }
              />
              <View
                style={
                  {
                    "borderRadius": 10,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 21,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Revolut
                </Text>
                <Text
                  style={
                    {
                      "color": "rgba(255,255,255,0.5)",
                      "fontSize": 10,
                      "marginBottom": 3,
                    }
                  }
                >
                  1234
                </Text>
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 18,
                    }
                  }
                >
                  £200
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "red",
                "borderRadius": 15,
                "height": 27,
                "justifyContent": "center",
                "opacity": 1,
                "position": "absolute",
                "right": 10,
                "top": 10,
                "width": 27,
              }
            }
            testID="close1"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={true}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderBottomColor": "gray",
                    "borderBottomWidth": 1,
                    "borderRadius": 20,
                    "elevation": 5,
                    "marginHorizontal": 20,
                    "marginVertical": "20%",
                    "padding": 40,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 4,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#000",
                      "fontSize": 22,
                      "fontWeight": "bold",
                      "marginBottom": 10,
                    }
                  }
                >
                  Remove Your Bank Account
                </Text>
                <View
                  style={
                    {
                      "backgroundColor": "lightgray",
                      "height": 1,
                      "marginVertical": 10,
                      "width": "100%",
                    }
                  }
                />
                <Text
                  style={
                    {
                      "color": "#000",
                      "fontSize": 18,
                      "marginVertical": 15,
                      "textAlign": "center",
                    }
                  }
                >
                  Are you sure you want to remove your bank account?
                </Text>
                <View
                  style={
                    {
                      "backgroundColor": "lightgray",
                      "height": 1,
                      "marginVertical": 10,
                      "width": "100%",
                    }
                  }
                />
                <View
                  style={
                    {
                      "alignItems": "flex-end",
                      "flexDirection": "row",
                      "justifyContent": "space-around",
                      "marginTop": 20,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "borderRadius": 10,
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        },
                        {
                          "backgroundColor": "#f44336",
                        },
                      ]
                    }
                    testID="pressable1"
                  >
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Yes
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "borderRadius": 10,
                          "paddingHorizontal": 20,
                          "paddingVertical": 10,
                        },
                        {
                          "backgroundColor": "#1E90FF",
                        },
                      ]
                    }
                    testID="pressable2"
                  >
                    <Text
                      style={
                        {
                          "color": "white",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      No
                    </Text>
                  </View>
                </View>
              </View>
            </Modal>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ecba1d",
                "borderRadius": 15,
                "bottom": 20,
                "height": 27,
                "justifyContent": "center",
                "opacity": 1,
                "position": "absolute",
                "right": 10,
                "width": 27,
              }
            }
            testID="close2"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 20,
                  },
                  {
                    "color": "white",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
